examples/UnderscoreOperator.kt

```diff
@@ -0,0 +1,30 @@
+// Taken from Kotlin 1.7
+// https://kotlinlang.org/docs/whatsnew17.html#underscore-operator-for-type-arguments
+
+abstract class SomeClass<T> {
+    abstract fun execute(): T
+}
+
+class SomeImplementation : SomeClass<String>() {
+    override fun execute(): String = "Test"
+}
+
+class OtherImplementation : SomeClass<Int>() {
+    override fun execute(): Int = 42
+}
+
+object Runner {
+    inline fun <reified S: SomeClass<T>, T> run(): T {
```

Comment:
Neither do these type parameters, though I'm actually not quite sure why. `tree-sitter-kotlin` seems to parse them when placed before the `fun` keyword, which isn't correct Kotlin though as per [the docs](https://kotlinlang.org/docs/generics.html#upper-bounds). So that's probably something we should fix in the grammar first too.